csfieldguide/templates/interactives/dictionary-compression.html

```diff
@@ -0,0 +1,84 @@
+{% extends interactive_mode_template %}
+
+{% load i18n %}
+{% load static %}
+
+{% block html %}
+  <div class="container">
+    <div class="row">
+      <div class="col s12">
+
+        <h3>{% trans "Compression using a Dictionary" %}</h3>
+
+        <div id="interactive-dictionary-compression">
+
+            <p>
+                {% trans "This interactive demonstrates the concept of using a dictionary to compress text." %}
+            </p>
+            <p>
+                {% trans "Using the input box below, enter a message. Then try and build a dictionary that compresses every character in the message." %}
+            </p>
+            <p>
+                {% trans "How efficient is your dictionary? Are you compressing one character per dictionary entry, or many? What happens if you add additional text to your message, does your dictionary still compress all of the text?" %}
+            </p>
+
+            <div id="interactive-dictionary-compression-activity">
+                <div id="interactive-dictionary-compression-left">
+                    <h2 class="interactive-dictionary-compression-heading">{% trans "Enter a message to compress" %}:</h2>
+                    <textarea id="interactive-dictionary-compression-input" placeholder="Pease porridge hot, pease porridge cold,
```

Comment:
It's being pattern-matched so, like the re-filter interactive I was just working on, I think the idea is that it's a specific string and translating that would change the resulting pattern. Worth a discussion probably because it's done in several places